src/sentry/tasks/embeddings_grouping/utils.py

```diff
@@ -172,7 +172,7 @@ def get_current_batch_groups_from_postgres(
         group_id_filter, project.id, batch_size
```

Comment:
`_make_postgres_call_with_retry` doesn't work properly afaict. This just constructs the queryset https://github.com/getsentry/sentry/blob/23768500359ae9535da4cd50ecb6c2a58f2d9ef6/src/sentry/tasks/embeddings_grouping/utils.py#L124-L133. But since we don't convert it to a list, or attempt to iterate over it, then it never runs the query.

That means you'll never enter your retry loop here: https://github.com/getsentry/sentry/blob/23768500359ae9535da4cd50ecb6c2a58f2d9ef6/src/sentry/tasks/embeddings_grouping/utils.py#L143-L149

You should coerce to a list in the return value. That also saves you the trouping of using `.count` here since you won't have a queryset.

Reply:
Thanks for the review @wedamija! I've opened up a follow up [PR](https://github.com/getsentry/sentry/pull/76081) to address these